src/lib.rs

```diff
@@ -94,7 +95,99 @@ fn serde_to_starlark<'v>(x: serde_json::Value, heap: &'v Heap) -> anyhow::Result
 
 // }}}
 
+// {{{ PythonObjectValue
+
+#[derive(Debug, ProvidesStaticType, NoSerialize, Allocative)]
+struct PythonObjectValue {
+    #[allocative(skip)]
+    obj: PyObject,
+}
+starlark_simple_value!(PythonObjectValue);
+
+impl Display for PythonObjectValue {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        Python::with_gil(|py| write!(f, "<python object: {}>", self.obj.bind(py).repr().unwrap()))
```

Comment:
The `unwrap()` call can lead to a runtime panic and should be avoided. I didn't look into it whether this unwrap can actually fail, but maybe https://pyo3.rs/v0.20.0/function/error_handling#foreign-rust-error-types gives some guidance.